plugins/module_utils/network/sonic/config/stp/stp.py

```diff
@@ -43,6 +49,44 @@
     {'pvst': {'vlan_id': ''}},
     {'rapid_pvst': {'vlan_id': ''}},
 ]
+
+
+def __derive_stp_root_config_delete_op(key_set, command, exist_conf):
+    done = True
+    new_conf = exist_conf
+    if command:
+        glbal = command.get('global', None)
+        if glbal and glbal.get('enabled_protocol', None):
+            new_conf.pop('interfaces', None)
+            new_conf.pop('mstp', None)
+            new_conf.pop('pvst', None)
+            new_conf.pop('rapid_pvst', None)
+            new_glbal = {'bpdu_filter': False,
+                         'bridge_priority': 32768,
+                         'fwd_delay': 15,
+                         'hello_time': 2,
+                         'loop_guard': False,
+                         'max_age': 20,
+                         'portfast': False}
+            new_conf['global'] = new_glbal
+        else:
+            done, new_conf = __DELETE_OP_DEFAULT(key_set, command, exist_conf)
+    return done, new_conf
+
+
+def __derive_stp_sub_config_delete_op(key_set, command, exist_conf):
+    done, new_conf = __DELETE_OP_DEFAULT(key_set, command, exist_conf)
+    done, new_conf = __DELETE_CONFIG_IF_NO_SUBCONFIG(key_set, command, new_conf)
+    return done, new_conf
+
+
+TEST_KEYS_generate_config = [
+    {'__default_ops': {'__delete_op': __derive_stp_root_config_delete_op}},
```

Comment:
__derive_stp_root_config_delete_op seems like it would still work as a default operation but name suggests moving under config key

Reply:
Changed. Thanks!